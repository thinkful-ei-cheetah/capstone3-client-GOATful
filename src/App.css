.page{
    animation-name: test;
    animation-duration: .5s;
}
body{
    pointer-events:none;
}
@keyframes test {
    0% {
        opacity: 0;
    }
    100%{
        opacity: 1;
    }
}
