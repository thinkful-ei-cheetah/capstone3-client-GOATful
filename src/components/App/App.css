.page{
    animation-name: test;
    animation-duration: .5s;
}
::-webkit-scrollbar { 
    display: none; 
}
@keyframes test {
    0% {
        opacity: 0;
    }
    100%{
        opacity: 1;
    }
}

.footer {
    background-color: rgba(36, 46, 77, 0.76);
    width: 100%;
    height: 3.5rem;
    color: white;
    display: flex;
    justify-content: space-between;
    align-items: center;
    bottom: 0;
    position: absolute;
  }

  .footer > span {
    padding: .75em;
  }

.footer a{
    color:rgb(203, 201, 201);
}

footer > span {
  padding: .75em;
}



@media screen and (max-width: 730px) {
    .footer {
        font-size: 12px;
    }

    footer div a:hover {
      font-size: 14px;
    }
}

.App {
    padding-bottom: 3.5rem;
  }
