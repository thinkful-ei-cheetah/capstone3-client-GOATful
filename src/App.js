import React , {Component} from 'react';
import { Route, Switch } from 'react-router-dom'
import PrivateRoute from './protected/PrivateRoute'
import PublicOnlyRoute from './protected/PublicOnlyRoute'


import NavBar from './components/NavBar/NavBar'
import Landing from './routes/Landing'
import AddVid from './routes/AddVideo'
import Creator from './routes/Creator'
import Login from './routes/Login'
import VidPrev from './routes/VidPreview'
import Videos from './routes/Videos'
import NotFound from './routes/NotFound'
import './App.css';


export default class App extends Component {
  state = {
    hasError : null,
  }

  render(){

    const landPage = () => {
      return(
        <Landing />
      )
    }

    const loginPage = () => {
      return(
        <Login />
      )
    }

    const creatorPage = () => {
      return(
        <Creator />
      )
    }

    const addVidPage = () => {
      return(
        <AddVid />
      )
    }

    const vidPrevPage = () => {
      return(
        <VidPrev />
      )
    }

    const videosPage = () => {
      return(
        <Videos />
      )
    }

    return(
      <div className="App">
        <NavBar />
        <main>
          <Switch>
            <PublicOnlyRoute
              exact
              path={'/'}
              component={landPage}
            />
            <Route
              exact
              path={'/creator'}
              component={creatorPage}
            />
            <PublicOnlyRoute
              exact
              path={'/add-video'}
              component={addVidPage}
            />
            <PublicOnlyRoute
              exact
              path={'/login'}
              component={loginPage}
            />
            <PrivateRoute 
              exact
              path={'/'}
              component={videosPage}
            />
            <PrivateRoute
              exact
              path={'/:vid-id/previews'}
              component={vidPrevPage}
            />
            <Route
              component={NotFound}
            />
          </Switch>
        </main>
      </div>
    )
  }
}

