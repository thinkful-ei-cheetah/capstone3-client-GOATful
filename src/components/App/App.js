import React, { Component } from 'react';
import { Route, Switch, Link } from 'react-router-dom'
import PrivateRoute from '../../protected/PrivateRoute'
import PublicOnlyRoute from '../../protected/PublicOnlyRoute'
import NavBar from '../NavBar/NavBar'
import LandingPage from '../../routes/LandingPage/LandingPage'
import TeamPage from '../../routes/MeetTeamPage/MeetTeamPage'
import AddVidPage from '../../routes/AddVideoPage/AddVideoPage'
import CreatorPage from '../../routes/CreatorPage/CreatorPage'
import LoginPage from '../../routes/LoginPage/LoginPage'
import PreviewsPage from '../../routes/PreviewsPage/PreviewsPage'
import VideosPage from '../../routes/VideosPage/VideosPage'
import NotFoundPage from '../../routes/NotFoundPage/NotFoundPage'
import './App.css';
import { withAppContext } from '../../contexts/AppContext'
import ErrorBar from '../ErrorBar/ErrorBar'
import Loader from '../Loader/Loader'

require('dotenv').config();


class App extends Component {
  state = {
    hasError: null,
  }

  renderErrorBar = () => {
    const { appError, clearAppError } = this.props.appContext
    if (appError) return <ErrorBar appError={appError} clearError={clearAppError} />
    return <></>
  }

  render() {
    return (
      <div className="App">
        <Loader />
        <NavBar />
        {this.renderErrorBar()}

        <main>
          <Switch>
            <PublicOnlyRoute
              exact path={'/'} component={LandingPage}
            />
            <Route
              exact path={'/creator'} component={CreatorPage}
            />
            <Route
              exact path={'/Team'} component={TeamPage}
            />
            <PublicOnlyRoute
              exact path={'/add-video'} component={AddVidPage}
            />
            <PublicOnlyRoute
              exact path={'/login'} component={LoginPage}
            />
            <PrivateRoute
              exact path={'/videos'} component={VideosPage}
            />
            <PrivateRoute
              exact path={'/videos/:video_id/previews'} component={PreviewsPage}
            />
            <Route
              component={NotFoundPage}
            />
          </Switch>
          <footer className='footer'>
            <span>Laconic: Better Thumbnails</span>
            <div>
              <Link
                to='/team'>
                Meet The Team
              </Link>
            </div>
            <span>Copyright GOATful 2019 </span>
          </footer>
        </main>
      </div>
    )
  }
}

export default withAppContext(App);