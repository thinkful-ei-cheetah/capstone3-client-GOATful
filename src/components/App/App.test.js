import React from 'react';
import App from './App';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import {BrowserRouter} from 'react-router-dom' 

describe('App Component', () => {

  // it('NavBar renders without crashing', () => {
  //     const wrapper = shallow(<App />)
  //     expect(toJson(wrapper)).toMatchSnapshot()
  // });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        < App />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><App/></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

})


