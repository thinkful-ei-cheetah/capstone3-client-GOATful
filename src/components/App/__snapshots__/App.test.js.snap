// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders the UI as expected 1`] = `
<div
  className="App"
>
  <header>
    <div
      className="title-container"
    >
      <a
        className="root-link"
        href="/"
        onClick={[Function]}
      >
        <svg
          className="logo"
          style={
            Object {
              "height": "40px",
              "width": "40px",
            }
          }
          version="1.1"
          viewBox="0 0 32 40"
          x="0px"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <title>
            shutter
          </title>
          <path
            d="M16 0.6c-8.498 0-15.4 6.903-15.4 15.4s6.903 15.4 15.4 15.4c8.498 0 15.4-6.903 15.4-15.4s-6.903-15.4-15.4-15.4zM15.611 1.602l1.098 0.053c4.354 0.211 8.206 2.377 10.699 5.627l0.244 0.32-4.316 7.393-7.176-12.439zM14.471 1.715l4.287 7.434h-15.418l0.598-0.926c2.229-3.447 5.888-5.881 10.135-6.453zM28.314 8.535l0.498 0.982c0.988 1.949 1.547 4.153 1.547 6.482 0 1.927-0.383 3.763-1.070 5.445l-0.152 0.373h-8.572l7.195-12.332zM2.863 10.182h8.578l-7.715 13.352-0.504-0.977c-1.015-1.969-1.59-4.2-1.59-6.557v-0.002c0.007-1.926 0.39-3.761 1.078-5.443zM12.645 10.184h6.719l3.379 5.846-3.375 5.783h-6.695l-3.387-5.816 3.186-5.514zM8.684 17.037l7.787 13.365-1.098-0.049c-4.361-0.191-8.226-2.34-10.734-5.576l-0.246-0.318 3.773-6.527zM13.277 22.857h15.395l-0.602 0.926c-2.221 3.42-5.852 5.847-10.064 6.439l-0.041 0.006h-0.391l-3.77-6.469z"
          />
          <text
            fill="#000000"
            fontFamily="'Helvetica Neue', Helvetica, Arial-Unicode, Arial, Sans-serif"
            fontSize="5px"
            fontWeight="bold"
            x="0"
            y="47"
          >
            Created by vectlab
          </text>
          <text
            fill="#000000"
            fontFamily="'Helvetica Neue', Helvetica, Arial-Unicode, Arial, Sans-serif"
            fontSize="5px"
            fontWeight="bold"
            x="0"
            y="52"
          >
            from the Noun Project
          </text>
        </svg>
        <h2
          className="app-title"
        >
          Laconic
        </h2>
      </a>
    </div>
    <nav
      className="stroke"
    >
      <a
        className="login link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
       
    </nav>
  </header>
  <main>
    <section
      className="hero page"
    >
      <h1>
        Let your thumbnail do the talking.
      </h1>
      <div>
        <h2
          className="red"
        >
          With Laconic
        </h2>
      </div>
    </section>
    <div
      className="landing-container"
    >
      <section
        className="click-through"
      >
        <figure>
          <img
            alt="Click through rate"
            src="https://www.lyfemarketing.com/blog/wp-content/uploads/2018/06/what-is-CTR.jpg"
          />
        </figure>
        <h3>
          So What Is Click-Through Rate, Anyway?
        </h3>
        <p>
          is the rate at which your videos are clicked. This number is the percentage of people who view your video (impressions) and then actually go on to click the video (clicks). The formula for CTR looks like this:
        </p>
        <h4>
          (Total Clicks on Video) / (Total Impressions) = Click Through Rate
        </h4>
        <p>
          Generally, you can view your click-thru rate within the dashboard of your YouTube account. A high CTR means that a high percentage of people who see your video click it.
        </p>
      </section>
      <section
        className="AB-testing"
      >
        <figure>
          <img
            alt="A/B Testing"
            src="https://mzdpd1mg35f29101q1lbu25g-wpengine.netdna-ssl.com/wp-content/uploads/2019/01/AB-guide-small.png"
          />
        </figure>
        <h3>
          WHAT IS A/B TESTING
        </h3>
        <p>
          An “A/B Test” is a test where 2 variants of something (anything) is tested for equal amounts of time, to see which version performs the best. In this case, what is being tested and compared is 2 versions of a thumbnail, of your own choosing. By uploading 2 thumbnails into the system, they are changed each day automatically on your channel (for a chosen period), and by the end of the test we automatically gather the data from your YouTube Analytics and see which of the thumbnails resulted in the most views.
        </p>
      </section>
      <section
        className="about"
      >
        <figure>
          <img
            alt="YouTube"
            src="https://images.unsplash.com/photo-1521302200778-33500795e128?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1500&q=80"
          />
        </figure>
        <h3>
          What we do
        </h3>
        <p>
          We Preview your thumbnails, so you can see what they will look like in Youtube's styling in both desktop and App view. Save your thumbnails so you can easily see the difference between the two. Keep all your thumbnail Ideas in one place!
        </p>
      </section>
      <div
        className="btn-container"
      >
        <button
          className="create-laconic"
          onClick={[Function]}
        >
          Create with Laconic
        </button>
      </div>
    </div>
    <footer
      className="footer"
    >
      <span>
        Laconic: Better Thumbnails
      </span>
      <div>
        <a
          href="/team"
          onClick={[Function]}
        >
          Meet The Team
        </a>
      </div>
      <span>
        Copyright GOATful 2019 
      </span>
    </footer>
  </main>
</div>
`;
